### ISSUE TITLE
False positive E1130 invalid-unary-operand-type for Numpy array

### ISSUE THREAD

**Issue by cdeil** [PyCQA/pylint#2436]
Pylint emits a false positive E1130 invalid-unary-operand-type error for Numpy arrays:
```
import numpy as np
a = np.array([1, 2])
b = -a
```
Actually, in this case, it emits two:
```
$ pylint test.py  -E
************* Module test
test.py:3:4: E1130: bad operand type for unary -: recarray (invalid-unary-operand-type)
test.py:3:4: E1130: bad operand type for unary -: list (invalid-unary-operand-type)
```

This is not just for np.array, I get a similar false positive for np.atleast_1d, which also returns Numpy arrays. But np.array is super common in scientific Python projects.

```
$ pylint --version
pylint 2.2.0
astroid 2.0.4
Python 3.6.0 | packaged by conda-forge | (default, Feb 10 2017, 07:08:35) 
[GCC 4.2.1 Compatible Apple LLVM 7.3.0 (clang-703.0.31)]
```

This issue was already mentioned in #1472 and #1497 , but @PCManticore you suggested to split out a new issue with a simple test case that doesn't involve flow control in https://github.com/PyCQA/pylint/issues/1472#issuecomment-355814591 , so there you have it.

:-)

Thanks for all your work on pylint, it's a huge help!

**Comment by PCManticore** [comment]
Thanks for creating an issue, can confirm the bug!

**Comment by NeilGirdhar** [comment]
This also triggers on scalar types like `numpy.float32`.

**Comment by Max16hr** [comment]
Bug is still there.
In my case `np.linspace` returns a `np.ndarray` object but when I try to negate that I will get a pylint error: 

> bad operand type for unary -: tuple

Obviously, this is a bug. There is no tuple at all.

**Comment by hippo91** [comment]
@cdeil in fact there are two problems:

- the first one, and easiest to solve, is that `astroid` (which is responsible for inference in `pylint`) doesn't recognize the unary operators `-` or `~` as valid operators for `numpy` types;
- the second one, and a little bit more complex, is that `astroid` infers the result of the function `array` (or `linspace` in @Max16hr's  case) as a `recarray`, which is correct, but also a `list` or a `tuple` depending on the type you passed as argument in the function.

Currently I have found a solution for both problems. I'm trying to make it the more general as possible.
